docs/taskruns.md

```diff
@@ -311,6 +311,23 @@ steps:
     startedAt: "2019-08-12T18:22:54Z"
   ```
 
+The following tables shows how to read the overall status of a `TaskRun`:
+
+`status`|`reason`|`completionTime` is set|TaskRun status
+:-------|:-------|:---------------------:|--------------:
+Unknown|Started|No|The TaskRun has just been picked up by the controller.
+Unknown|Pending|No|The TaskRun is waiting on a Pod in status Pending.
+Unknown|Running|No|The TaskRun has been validate and started to perform its work.
+Unknown|TaskRunCancelled|No|The user requested the TaskRun to be cancelled. Cancellation has not be done yet.
+True|Succeeded|Yes|The TaskRun completed successfully.
+False|Failed|Yes|The TaskRun failed because one of the steps failed.
+False|\[Error message\]|No|The TaskRun encountered an error, but it's still running.
+False|\[Error message\]|Yes|The TaskRun failed with a permanent error (usually validation).
+False|TaskRunCancelled|Yes|The TaskRun was cancelled successfully.
+False|TaskRunTimeout|Yes|The TaskRun timed out.
+
+When a `TaskRun` changes status, [events](events.md#taskruns) are triggered accordingly.
+
```

Comment:
Similar observation as before, may be add a little more description to "The TaskRun encountered an error, but it's still running.", change it "The TaskRun encountered an error, but it's still running since waiting for other tasks or steps (need to check 🔍 ) to finish executing." 🤔

Reply:
Updated, this is again about transient errors

Reply:
something like?

```
{"level":"info","logger":"tekton.taskrun-controller.event-broadcaster","caller":"record/event.go:274","msg":"Event(v1.ObjectReference{Kind:\"TaskRun\", Namespace:\"default\", Name:\"pipelinerun-one-failure-two-success-task-c-9fmnd\", UID:\"a788c615-9058-45ae-8dea-de1310343208\", APIVersion:\"tekton.dev/v1beta1\", ResourceVersion:\"4190777\", FieldPath:\"\"}): type: 'Warning' reason: 'Error' Operation cannot be fulfilled on taskruns.tekton.dev \"pipelinerun-one-failure-two-success-task-c-9fmnd\": the object has been modified; please apply your changes to the latest version and try again","commit":"ac439fb","knative.dev/controller":"taskrun-controller"}
```

Reply:
Yep, failures on update are the only case. Hopefully with the genreconciler we'll see less of that